### PR TITLE
switch from dorny/test-reporter to test-summary/action

### PR DESCRIPTION
**Pull request by wicksipedia**
- test results will be included in the job summary
- simplifies actions
- requires use of junit output https://github.com/test-summary/action/issues/1
- added publish and upload artifact steps
- tested failure case by committing with failing test to ensure it works when tests fail (then reverted failing test)

Test report is now included as a summary instead of a separate job (eliminates potential issue of report being attached to wrong workflow)
CleanShot 2023-12-13 at 14 33 16

GitHub templates/workflows now included in template
CleanShot 2023-12-13 at 14 27 34

### CODE REVIEW

File: tests/Directory.Packages.props [hunk 1]
```diff
@@ -1,23 +1,22 @@
...
-        <PackageVersion Include="NSubstitute" Version="5.1.0" />
-        <PackageVersion Include="Respawn" Version="6.1.0" />
-        <PackageVersion Include="Testcontainers" Version="3.6.0" />
-        <PackageVersion Include="Testcontainers.SqlEdge" Version="3.6.0" />
-        <PackageVersion Include="Microsoft.AspNetCore.Mvc.Testing" Version="8.0.0" />
-        <PackageVersion Include="Microsoft.Extensions.DependencyInjection.Abstractions" Version="8.0.0" />
-        <PackageVersion Include="Meziantou.Extensions.Logging.Xunit" Version="1.0.7" />
-    </ItemGroup>
-</Project>
+  <PropertyGroup>
```
Comment (danielmackay) [inline tests/Directory.Packages.props:2]: Is there a standard about using 2 spaces for `xml` files?

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -36,12 +32,26 @@ jobs:
       run: dotnet build --no-restore --configuration Release
 
     - name: Test
-      run: dotnet test --no-build --no-restore --configuration Release --logger "trx;LogFileName=TestResults.trx"
+      run: dotnet test --no-build --no-restore --configuration Release --logger "junit;LogFilePath=test-result.xml"
 
-    # as per https://github.com/dorny/test-reporter?tab=readme-ov-file#recommended-setup-for-public-repositories
-    - name: Test Report
+    - name: Publish
```
Comment (danielmackay) [inline .github/workflows/build-and-test.yml:37]: Why do we need to publish & upload artifacts on build & test?

**Review by danielmackay — COMMENTED** [review]
2 comments.

File: .github/workflows/build-and-test.yml (hunk 2 above)
Comment (wicksipedia) [reply .github/workflows/build-and-test.yml:37]: so you have an artifact that can be deployed

File: tests/Directory.Packages.props (hunk 1 above)
Comment (wicksipedia) [reply tests/Directory.Packages.props:2]: this is how the `.editorconfig` is configured

**Review by danielmackay — APPROVED** [review]
LGTM

**Pull request merged**